sed/io.py

```diff
@@ -75,89 +113,97 @@ def res_to_xarray(res, bin_names, bin_axes, metadata=None):
 
     for name in bin_names:
         try:
-            xres[name].attrs['unit'] = DEFAULT_UNITS[name]
+            xres[name].attrs["unit"] = DEFAULT_UNITS[name]
         except KeyError:
             pass
```

Comment:
if we don't handle the error, why are we using a `try...except` block?

Reply:
This part is anyways gone now, but there are similar places: The idea is to silently skip over missing items in the dictionary. How would you do this better? Check with if clauses for the keys first?

Reply:
Yep, and "if not ..." should be ok